src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidTestCommand.cs

```diff
@@ -120,6 +143,36 @@ protected override Task<ExitCode> InvokeInternal(ILogger logger)
             return Task.FromResult(ExitCode.GENERAL_FAILURE);
         }
 
+        private string GetDeviceToUse(ILogger logger, AdbRunner runner, string apkRequiredArchitecture)
+        {
+            var allDevicesAndTheirArchitectures = runner.GetAttachedDevicesAndArchitectures();
+            if (allDevicesAndTheirArchitectures.Count == 1)
+            {
+                // There's only one device, so -s argument isn't needed but still check
+                if (!allDevicesAndTheirArchitectures.Values.Contains(apkRequiredArchitecture, StringComparer.InvariantCultureIgnoreCase))
+                {
+                    logger.LogError($"Single device available has architecture '{allDevicesAndTheirArchitectures.Values.First()}', != '{apkRequiredArchitecture}'. Package installation will likely fail, but we'll try anyways."); 
+                }
+                return ""; // Empty = Use default device
+            }
+            else if (allDevicesAndTheirArchitectures.Count > 1)
+            {
+                if (allDevicesAndTheirArchitectures.Any(kvp => kvp.Value.Equals(apkRequiredArchitecture, StringComparison.OrdinalIgnoreCase)))
+                { 
+                    var firstAvailableCompatible = allDevicesAndTheirArchitectures.Where(kvp => kvp.Value.Equals(apkRequiredArchitecture, StringComparison.OrdinalIgnoreCase)).FirstOrDefault();
+                    logger.LogDebug($"Using first-found compatible device - serial: '{firstAvailableCompatible.Key}' - Arch: {firstAvailableCompatible.Value}");
```

Comment:
I think it might be helpful to say how many other devices were there. I can see how that can help debugging in the future when you have three devices connected, one of them dead and it only detects two but you will never know or something like that.
(just what I also observed as a custom in Xamarin so there might be value in it)